### PR TITLE
auto upload cuda12.1 python pkg to release when create new tag

### PR DESCRIPTION
**Pull request by irexyc**
## Motivation

address https://github.com/InternLM/lmdeploy/issues/781

https://github.com/irexyc/lmdeploy/releases

### CODE REVIEW

File: .github/workflows/cuda12.1-whl-release.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    tags:
+      - '*'
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+  linux-build:
+    strategy:
```
Comment (RunningLeon) [inline .github/workflows/cuda12.1-whl-release.yml:14]: could we use matrix strategy and merge them togather in https://github.com/InternLM/lmdeploy/blob/d3e2cee41e75bda8a645c521e8b06e8f29b84e47/.github/workflows/pypi.yml#L15?

File: .github/workflows/cuda12.1-whl-release.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+          echo ${DOCKER_TAG}
+          echo ${OUTPUT_FOLDER}
+          # remove -it
+          sed -i 's/docker run --rm -it/docker run --rm/g' builder/manywheel/build_wheel.sh
+          bash builder/manywheel/build_wheel.sh ${PYTHON_VERSION} ${PLAT_NAME} ${DOCKER_TAG} ${OUTPUT_FOLDER}
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          if-no-files-found: error
+          path: builder/manywheel/${{ env.OUTPUT_FOLDER }}/*
```
Comment (RunningLeon) [inline .github/workflows/cuda12.1-whl-release.yml:52]: can we update a file with its parent directory?

File: .github/workflows/cuda12.1-whl-release.yml (hunk 1 above)
Comment (irexyc) [reply .github/workflows/cuda12.1-whl-release.yml:14]: 感觉分开比较好， pypi也只是上传cuda11.8的部分，两个ci之间也没有重叠的部分。

**Review by lvhan028 — APPROVED** [review]
LGTM

**Review by RunningLeon — APPROVED** [review]
LGTM after fixing the lint error

**Pull request merged**